newbank/server/NewBankClientHandler.java

```diff
@@ -72,7 +75,27 @@ public void run() {
 		}
 	}
 
-	public CustomerID login() throws IOException {
+	public CustomerID startUp() throws IOException {
+		boolean startUp = true;
+		try {
```

Comment:
Hi @Georgeh712 just a few things I want to note:
1. Do you think we should abstract the register method, as well as the login method instead of having them inline like this? The main benefit of this is that it makes the code more readable. Also, decoupling login from register (and then calling the relevant one in startup) means that we can separate one from the other and perhaps change how they are used independently
2. As a follow-up to that. It seems a user will have to login again after registering. That seems a little weird, do you think? Because now I have to enter the same username (and password when that is implemented) twice.
<img width="279" alt="image" src="https://user-images.githubusercontent.com/33374480/159172555-804497ee-2155-4c0d-ba26-5e1db26b0f05.png">

Reply:
Yes, I see your point in both cases, let me have a go.